### DISCUSSION TITLE
user defined functions and targets

### DISCUSSION THREAD

**Discussion by c1au6i0** [ropensci/targets#998] [Help] [answered]
Is there a place, beside `workspaces` where `targets` keep the user defined functions called by the different targets?
Just realized that I delete a function before committing the changes and I was figuring out if I could retrieve it somewhere.

Thank you
Claudio

**Comment by wlandau** [comment, marked as answer]
`targets` stores the hashes of global objects and functions, but not the objects of functions themselves. To retrieve the function, you could look in your commit history if your project code is under version control.
